src/cmd-tag

```diff
@@ -0,0 +1,211 @@
+# Allows users to operate on the tags in `builds.json`
+#
+# Examples:
+#
+#   $ coreos-assembler tag create --tag stable --build 47.119
+#   $ coreos-assembler tag update --tag smoketested --build 47.152
+#   $ coreos-assembler tag delete --tag beta
+#   $ coreos-assembler tag list
+#
+import argparse
+import json
+import os
+import sys
+
+sys.path.insert(0, '/usr/lib/coreos-assembler')
+from cmdlib import fatal, rfc3339_time, write_json
+
+
+def main():
+    """Main entry point"""
+
+    args = parse_args()
+    args.func(args)
+
+
+def parse_args():
+    """Parse args and dispatch"""
+
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest="cmd", title="Tag actions")
+    subparsers.required = True
+
+    create = subparsers.add_parser("create", help="Create a tag from a build")
+    create.add_argument("--tag", help="Tag to be created", required=True)
+    create.add_argument("--build", help="Build to be tagged", required=True)
+    create.add_argument("--force", help="Force tag creation",
+                        action="store_true")
+    create.set_defaults(func=cmd_create)
+
+    delete = subparsers.add_parser("delete",
+                                   help="Delete a tag from the metadata")
+    delete.add_argument("--tag", help="Tag to delete from metadata",
+                        required=True)
+    delete.set_defaults(func=cmd_delete)
+
+    list_tags = subparsers.add_parser("list", help="List available tags in "
+                                                   "the metadata")
+    list_tags.set_defaults(func=cmd_list)
+
+    update = subparsers.add_parser("update", help="Update a tag to new build")
+    update.add_argument("--tag", help="Tag to be updated", required=True)
+    update.add_argument("--build", help="Build to update tag with",
+                        required=True)
+    update.add_argument("--force", help="Force the update of a tag",
+                        action="store_true")
+    update.set_defaults(func=cmd_update)
+
+    return parser.parse_args()
+
+
+def init_build_data():
+    """Initialize build metadata"""
+
+    builds_json = "builds/builds.json"
+
+    if os.path.isfile(builds_json):
+        with open(builds_json) as f:
+            builddata = json.load(f)
+    else:
+        print("INFO: Did not find existing builds.json; starting with empty "
+              "build list and tags list")
+        builddata = {"builds": [], "tags": []}
+
+    return builddata
+
+
+def finalize_json(builddata):
+    """Wrapper around cmdlib.write_json()"""
+
+    builds_json = "builds/builds.json"
+
+    builddata['timestamp'] = rfc3339_time()
+    write_json(builds_json, builddata)
+
+
+def available_builds(builddata):
+    """Returns the availble build IDs from build metadata"""
+
+    return builddata["builds"]
+
+
+def available_tags(builddata):
+    """Returns the available tag names from build metadata"""
+
+    available_tag_names = []
+    if "tags" in builddata.keys():
```

Comment:
Minor: should be able to just do `if "tags" in builddata`.

Edit: and similar in below occurrences of `in builddata.keys()`

Reply:
:+1:

Reply:
This whole function could be simplified to:
```
def available_tags(builddata):
    return [t.get("name") for t in builddata.get("tags")]
```

Reply:
Updated in next commit.